index.js

```diff
@@ -12,7 +12,26 @@ function isNegative(pattern) {
 	return pattern[0] === '!';
 }
 
+function isString(value) {
+	return typeof value === 'string';
+}
+
+function assertPatternsInput(patterns) {
+	if (isString(patterns)) {
+		// bail early, the rest will easier this way
+		return;
+	}
+
+	if (!Array.isArray(patterns) || !patterns.every(isString)) {
+		throw new TypeError('patterns must be a string or an array of strings');
+	}
+}
+
 function generateGlobTasks(patterns, opts) {
+	// make sure input is correct, so that we can safely parse
+	// it in this task
+	assertPatternsInput(patterns);
+
 	var globTasks = [];
 
 	patterns = arrify(patterns);
```

Comment:
Excellent points, @schnittstabil !

@catdad Would you like to apply these changes or do you prefer us to do so?